server/src/main/java/io/crate/protocols/postgres/types/DateType.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.protocols.postgres.types;
+
+import javax.annotation.Nonnull;
+import java.nio.charset.StandardCharsets;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.time.format.ResolverStyle;
+import java.util.Locale;
+
+import static java.time.format.DateTimeFormatter.ISO_LOCAL_DATE;
+
+
+final class DateType extends BaseTimestampType {
+
+    public static final PGType INSTANCE = new DateType();
+
+    private static final int OID = 1082;
+    private static final String NAME = "date";
+
+    private static final DateTimeFormatter ISO_FORMATTER = new DateTimeFormatterBuilder()
+        .parseCaseInsensitive()
+        .append(ISO_LOCAL_DATE)
+        .toFormatter(Locale.ENGLISH).withResolverStyle(ResolverStyle.STRICT);
+
+    private static final DateTimeFormatter ISO_FORMATTER_AD = new DateTimeFormatterBuilder()
+        .parseCaseInsensitive()
+        .appendPattern("yyyy-MM-dd")
+        .toFormatter(Locale.ENGLISH).withResolverStyle(ResolverStyle.STRICT);
+
+    private DateType() {
+        super(OID, TYPE_LEN, TYPE_MOD, NAME);
+    }
+
+    @Override
+    public int typArray() {
+        return PGArray.DATE_ARRAY.oid();
+    }
+
+    @Override
+    byte[] encodeAsUTF8Text(@Nonnull Object value) {
+        long millis = (long) value;
+        LocalDate date = LocalDate.ofInstant(Instant.ofEpochMilli(millis), ZoneOffset.UTC);
+
+        return date.format(ISO_FORMATTER_AD).getBytes(StandardCharsets.UTF_8);
+    }
+
+    @Override
+    Object decodeUTF8Text(byte[] bytes) {
+        String s = new String(bytes, StandardCharsets.UTF_8);
+
+        LocalDate dt = LocalDate.parse(s, ISO_FORMATTER);
+        return dt.atStartOfDay(ZoneOffset.UTC).toInstant().toEpochMilli();
```

Comment:
PostgreSQL supports more text formats as input:

https://www.postgresql.org/docs/current/datatype-datetime.html

I think we either have to extend this, or make it explicit in the documentation that the text format support is not compatible with all PostgreSQL clients.

Reply:
I updated `data-types.rst` and added TODO to `DateType`